x-pack/filebeat/input/netflow/decoder/config/config.go

```diff
@@ -30,7 +30,7 @@ type Config struct {
 
 var defaultCfg = Config{
 	protocols:       []string{},
-	logOutput:       io.Discard,
+	logOutput:       logp.L().Named("netflow"),
```

Comment:
The godoc for `Defaults()` says "log output is discarded".

With this line, it is hard to guarantee that is true because initialization order can affect `logp.L()`. The global logger may or may not be initialized at the time at which this call is executed.

I didn't look at how this is used, but maybe it's better to separate the logger from the decoder config concept. And require that users of the decoders pass their own parent `*logp.Logger` at construction time.